gcc/testsuite/rust/compile/builtin_macro_concat.rs

```diff
@@ -6,12 +6,13 @@ macro_rules! concat {
 fn main() {
     let not_literal = "identifier";
     concat!();
-    concat! (,); // { dg-error "argument must be a constant literal" }
-    concat!(not_literal); // { dg-error "argument must be a constant literal" }
+    concat! (,); // { dg-error "expected expression, found .,." }
+    concat!(not_literal); // { dg-error "expected a literal" }
     concat!("message");
     concat!("message",);
     concat!("message", 1, true, false, 1.0, 10usize, 2000u64);
     concat!("message", 1, true, false, 1.0, 10usize, 2000u64,);
-    concat! ("m", not_literal); // { dg-error "argument must be a constant literal" }
-    concat!(not_literal invalid 'm' !!,); // { dg-error "argument must be a constant literal" }
+    concat! ("m", not_literal); // { dg-error "expected a literal" }
+    concat!(not_literal invalid 'm' !!,); // { dg-error "expected token: .,." }
+                                          // { dg-error "expected a literal" "" { target *-*-* } .-1 }
```

Comment:
Note that this test case was changed so that it now expects the diagnostics that follow `rustc`'s diagnostics more closely (I'd say 90% similarity).